### ISSUE TITLE
about loss function

### ISSUE THREAD

**Issue by shaoxuying** [herandy/DEPICT#1]
It seems that the loss function in your code is  different from that in your paper. I've changed into the loss you mentioned in your paper, it didn't work very well.  Could you give me some advice?

**Comment by KamranGhasedi** [comment]
Could you please let me know which lines you found different?

**Comment by shaoxuying** [comment]
line827 in functions.py
https://github.com/herandy/DEPICT/blob/master/functions.py

**Comment by herandy** [comment]
If you look at equation 12 in the paper, also line 3 in the algorithm, the second part of the loss function which contains z_i actually contains what we are doing at line 827, using p_i instead of z_i should be roughly equivalent.

**Comment by KamranGhasedi** [comment]
Considering any hyper-parameter for KL(f||u) in eq. (4) is equivalent of power in nominator of eq. (7). We set this hyper-parameter equal to 2 for all datasets. Intuitively,  setting higher weight for KL(f||u)  in the loss function is more helpful for the datasets with uniform class labels.
